sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -2360,7 +2360,10 @@ class AstBuilder extends SqlBaseParserBaseVisitor[AnyRef] with SQLConfHelper wit
       Literal(converter(rawStrippedQualifier))
     } catch {
       case e: NumberFormatException =>
-        throw new ParseException(e.getMessage, ctx)
+        throw new ParseException(
```

Comment:
We should also mention it in readme.md or somewhere: we don't have to instantiate the exception instance in a central file if it's only used once. This can also simplify the error migration work.

Reply:
> we don't have to instantiate the exception instance in a central file if it's only used once.

Don't think we should make this excuse for such case. Wenchen, the purpose of the migration on the temporary error classes is to do that with as less effort as possible. That's why I didn't move the exception to a central file.